docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -185,25 +186,16 @@ And the following defaulting:
 
 - `KubeadmControlPlane.Spec.Replicas: 1`
 
-#### Modifications required to existing API Types
-
-- Add `Cluster.Spec.ControlPlaneRef` defined as:
+##### Changes for supporting different upgrade strategies
 
 ```go
-    // ControlPlaneRef is an optional reference to a provider-specific resource that holds
-    // the details for provisioning the Control Plane for a Cluster.
-    // +optional
-    ControlPlaneRef *corev1.ObjectReference `json:"controlPlaneRef,omitempty"`
-```
-
-- Add `Cluster.Status.ControlPlaneReady` defined as:
+type StrategyType string
 
-```go
-    // ControlPlaneReady defines if the control plane is ready.
-    // +optional
-    ControlPlaneReady bool `json:"controlPlaneReady,omitempty"`
+const (
+  // ScaleInStrategyType constant is to set strategy type to scale in.
+  ScaleInStrategyType StrategyType = "ScaleIn"  
```

Comment:
There should be also ScaleOutStrategyType (for completeness and also because nothing prevents the user to explicitly ask for this strategy, even if this is the default).